cpp/ql/src/Security/CWE/CWE-416/UseOfStringAfterLifetimeEndsBad.cpp

```diff
@@ -0,0 +1,7 @@
+#include <string>
+void work(const char*);
+
```

Comment:
We normally include a [comment here](https://codeql.github.com/codeql-query-help/cpp/cpp-dangerous-cin/). Something like this, but I'm sure you can write this more clearly:

```suggestion

// BAD: `c_str` is destroyed on return so `work` acts on a pointer to invalid memory

```

Reply:
Many thanks. When I checked the qhelp style guide, I was surprised to see this wasn't mentioned there. Indeed, the C qhelp files don't follow this convention consistently, but this is more consistent in other languages.